optimum/intel/openvino/quantization.py

```diff
@@ -0,0 +1,223 @@
+        model_type = self.model.config.model_type.replace("_", "-")
+        onnx_config_cls = FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][self.feature]
+        onnx_config = onnx_config_cls(self.model.config)
+        compressed_model.eval()
+        use_external_data_format = onnx_config.use_external_data_format(compressed_model.num_parameters())
+        f = io.BytesIO() if not use_external_data_format else output_path.replace(".xml", ".onnx")
+
+        # Export the compressed model to the ONNX format
+        self._onnx_export(compressed_model, onnx_config, model_inputs, f)
+
+        # Load and save the compressed model
+        model = core.read_model(f) if use_external_data_format else core.read_model(f.getvalue(), b"")
+        self._save_pretrained(model, output_path)
+
+    @staticmethod
+    def _save_pretrained(model: openvino.runtime.Model, output_path: str):
+        pass_manager = passes.Manager()
+        pass_manager.register_pass("Serialize", output_path, output_path.replace(".xml", ".bin"))
+        pass_manager.run_passes(model)
+
+    @staticmethod
+    def _onnx_export(model: torch.nn.Module, config: OnnxConfig, model_inputs: Dict, f: Union[str, io.BytesIO]):
+        with torch.no_grad():
+            # Disable node additions to be exported in the graph
+            model.disable_dynamic_graph_building()
+            onnx_export(
+                model,
+                tuple(model_inputs.values()),
+                f=f,
+                input_names=list(model_inputs.keys()),
+                output_names=list(config.outputs.keys()),
+                dynamic_axes={name: axes for name, axes in chain(config.inputs.items(), config.outputs.items())},
+                do_constant_folding=True,
+                opset_version=10,
+            )
+            model.enable_dynamic_graph_building()
+
+    def get_calibration_dataset(
+        self,
+        dataset_name: str,
+        num_samples: int = 100,
+        dataset_config_name: Optional[str] = None,
+        dataset_split: Optional[str] = None,
+        preprocess_function: Optional[Callable] = None,
+        preprocess_batch: bool = True,
+        use_auth_token: bool = False,
+    ) -> Dataset:
+        """
+        Create the calibration `datasets.Dataset` to use for the post-training static quantization calibration step.
+
+        Args:
+            dataset_name (`str`):
+                The dataset repository name on the Hugging Face Hub or path to a local directory containing data files
+                to load to use for the calibration step.
+            num_samples (`int`, defaults to 100):
+                The maximum number of samples composing the calibration dataset.
+            dataset_config_name (`str`, *optional*):
+                The name of the dataset configuration.
+            dataset_split (`str`, *optional*):
+                Which split of the dataset to use to perform the calibration step.
```

Comment:
Is there a default? If a dataset has train/val splits and I don't specify a split, will the entire dataset be used, or will I get an error?

Reply:
No in this case there will be an error, and because in many cases there is a train/eval split it makes sense to add the training set as default, I will add this modification